File: mtc-model/src/test/scala/org/drugis/mtc/ResultsUtil.scala
package org.drugis.mtc

object ResultsUtil {
	def getSamples(r: MCMCResults, p: Int, c: Int): Array[Double] = {
		(0 until r.getNumberOfSamples).map(
			i => r.getSample(p, c, i)).toArray
	}
	def getSamples(r: MCMCResults, p: Parameter, c: Int): Array[Double] = {
		getSamples(r, r.findParameter(p), c)
	}
}

File: mtc-data/src/test/java/org/drugis/mtc/ResultsUtil.java
package org.drugis.mtc;

public class ResultsUtil {
	public static double[] getSamples(MCMCResults r, int p, int c) {
		double[] samples = new double[r.getNumberOfSamples()];
		for (int i = 0; i < r.getNumberOfSamples(); ++i) {
			samples[i] = r.getSample(p, c, i);
		}
		return samples;
	}

	public static double[] getSamples(MCMCResults r, Parameter p, int c) {
		return getSamples(r, r.findParameter(p), c);
	}
}
